src/components/search/data/tests/searchLoader.test.jsx

```diff
@@ -0,0 +1,121 @@
+import { screen, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { redirect } from 'react-router-dom';
+
+import { when } from 'jest-when';
+import { renderWithRouterProvider } from '../../../../utils/tests';
+import {
+  extractEnterpriseCustomer, queryAcademiesList,
+} from '../../../app/data';
+
+import { ensureAuthenticatedUser } from '../../../app/routes';
+import { enterpriseCustomerFactory } from '../../../app/data/services/data/__factories__';
+import makeSearchLoader from '../searchLoader';
+
+const mockEnterpriseCustomer = enterpriseCustomerFactory({
+  enableOneAcademy: true,
```

Comment:
nit: I'm assuming most enterprise customers will have `enableOneAcademy: false` in their configuration? If that's the case, should the default be false and have your test case that assumes this config is truthy override the default mock returning `enableOneAcademy: false` instead?

nit x2: The options passed to the `enterpriseCustomerFactory` should mimic the API response (i.e., using snake_case). I would recommend adding `enable_one_academy: false` to the `enterpriseCustomerFactory`'s definition (`Factory.define('enterpriseCustomer')`), and then passing `enable_one_academy: true` when you want to override the default value.